library/src/main/scala/scala/scalajs/macroimpls/UseAsMacros.scala

```diff
@@ -21,6 +21,7 @@ import Compat210._
  *
  *  @author Tobias Schlatter
  */
+@deprecated("Not actually deprecated, makes warnings go away", "")
 private[scalajs] object UseAsMacros {
```

Comment:
I have manually checked that using these macros does not actually trigger a deprecation warning. However, we currently do not have an automated mechanism that checks this (could change in future Scala versions).

A quick and dirty fix would be to abuse one of the example projects as a test (since they get compiled under `-XfatalWarnings` unlike the tests).